### PR TITLE
Added 'line' chart type

### LINKED ISSUE #1727: [Enhancement] Basic Line type Chart Component
- [X] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
The [new chart component](https://cookbook.kirby.design/home/showcase/chart) should be extended to support line charts.

## Describe the solution you'd like
A screenshot of the desired line chart has been attached. It should simply be implemented as a new 'type' for the chart component, that displays the supplied data as a line. It should also be possible to have several lines by supplying several datasets to the chart.

The marked area in the background is not needed for now. It should just be possible to draw 3 separate lines with 3 different colours. 

The implementation of the labels are handled in this issue: #1729 

image

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](../CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [X] Request that the issue is [UX refined](../CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [X] Request that the issue is [tech refined](../CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [X] Make sure you have read: "[Before you get coding](../CONTRIBUTING.md/#before-you-get-coding)".
- [X] Signal to others you are working on the issue by assigning yourself.
- [X] Create a branch from the [master branch](https://github.com/kirbydesign/designsystem/tree/master) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### PR DESCRIPTION
**Pull request by MadsBuchmann**
## Which issue does this PR close?

This PR closes #1727 

## What is the new behavior?
A new 'line' type has been added to the chart component. This results in the given dataset being drawn as a line chart. 

I've also updated the code for `chartJSService.updateType` method as it was not able to handle the update to `line` if the chart was previously a `bar` or `column` chart, as it should happen non-destructively. I.E. without destroying the existing chart element. 

Besides this, this is the first time I've really cracked open the files since the new chart.js charts was merged to master. 
So looking at it with fresh eyes I found some ample opportunities for refactoring, I've tried summarising them here: 
- A ChartConfigService has been added in order to be able to mock out configs during testing. 
- Types has been added to CHART_ANNOTATIONS_CONFIG & CHART_TYPES_CONFIG
- Config related files have been moved & split up into new files in the ../config folder 
- Some of the tests has been moved to `chart-js.service.spec.integration.ts` as these test rely on the actual configs and not the mocked versions. In other words, they test the styling of the chart.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](../CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [X] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [X] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [X] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [X] Request that the changes are code-reviewed 
- [X] Request that the changes are [UX reviewed](./CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be automatically merged to master via [automerge](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/automatically-merging-a-pull-request).

### CODE REVIEW

File: libs/designsystem/src/lib/components/chart/configs/type.config.ts [hunk 1]
```diff
@@ -117,27 +68,37 @@ export const CHART_TYPE_CONFIGS = {
...
+      elements: {
+        point: {
+          radius: 0,
         },
       },
     },
   },
 };
 
+/* TODO: Find a proper place to store these two */
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/chart/configs/type.config.ts:92]: Remember todo :)

**Review by RasmusKjeldgaard — DISMISSED** [review]
Love it! ❤️ 

LGTM

File: libs/designsystem/src/lib/components/chart/chart-js/chart-js.service.integration.spec.ts [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+      let chart: Chart;
+      it('should use correct ChartJS type', () => {
+        chartJSService.renderChart({
+          targetElement: canvasElement,
+          type: 'column',
+          data: [1, 2, 3],
+        });
+        const chart = chartJSService['chart'];
+
+        expect(chart.config.type).toBe('bar');
```
[the remark this answers is not in the document]
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/chart/chart-js/chart-js.service.integration.spec.ts:39]: A comment has been added :D

File: libs/designsystem/src/lib/components/chart/chart-js/chart-js.service.integration.spec.ts [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+          expect(chart.options.elements.line.hoverBackgroundColor).toBeUndefined();
+        });
+
+        // TODO: Figure out typography for chart
+        /*it('should be rendered with correct typography for data labels', () => {
+          const { size } = chart.options.scales['y'].ticks.font as FontSpec;
+          const { color } = chart.options;
+          expect(size).toBe(14);
+          expect(color).toBe(ColorHelper.getThemeColorHexString('black'));
+        });*/
```
[the remark this answers is not in the document]
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/chart/chart-js/chart-js.service.integration.spec.ts]: Accidentally left it in - it has been solved instead :)

File: libs/designsystem/src/lib/components/chart/configs/type.config.ts (hunk 1 above)
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/chart/configs/type.config.ts:106]: I've found a proper place to store these two! 
Todo resolved :D

**Review by RasmusKjeldgaard — APPROVED** [review]

**Pull request merged**